core/01_core.js

```diff
@@ -127,6 +127,18 @@
     errorMap[className] = errorBuilder;
   }
 
+  function buildCustomError(className, message, code) {
+    const error = errorMap[className]?.(message);
+    // Strip buildCustomError() calls from stack trace
+    if (typeof error == "object") {
+      ErrorCaptureStackTrace(error, buildCustomError);
+      if (code) {
+        error.code = code;
+      }
+    }
+    return error;
```

Comment:
Nit: minor, but it can be clearer, (when isn't the result an object?)
```suggestion
    const errorConstructor = errorMap[className];
    if (errorConstructor !== undefined) {
      const error = errorConstructor(message);
      // Strip buildCustomError() calls from stack trace
      ErrorCaptureStackTrace(error, buildCustomError);
      if (code) {
        error.code = code;
      }
      return error;
    }
```

Reply:
Theoretically the error constructor function could return anything, eg. null, I guess.